compiler/noirc_frontend/src/parser/parser/generics.rs

```diff
@@ -0,0 +1,295 @@
+            return None;
+        };
+
+        if !self.eat_colon() {
+            self.push_error(
+                ParserErrorReason::MissingTypeForNumericGeneric,
+                self.current_token_span,
+            );
+            return Some(UnresolvedGeneric::Numeric { ident, typ: type_u32() });
+        }
+
+        let typ = self.parse_type_or_error();
+        if let UnresolvedTypeData::Integer(signedness, bit_size) = &typ.typ {
+            if matches!(signedness, Signedness::Signed)
+                || matches!(bit_size, IntegerBitSize::SixtyFour)
+            {
+                self.push_error(ParserErrorReason::ForbiddenNumericGenericType, typ.span);
+            }
+        }
+
+        Some(UnresolvedGeneric::Numeric { ident, typ })
+    }
+
+    /// ResolvedGeneric = quoted_type
+    fn parse_resolved_generic(&mut self) -> Option<UnresolvedGeneric> {
+        if let Some(token) = self.eat_kind(TokenKind::QuotedType) {
+            match token.into_token() {
+                Token::QuotedType(id) => {
+                    return Some(UnresolvedGeneric::Resolved(id, self.previous_token_span));
+                }
+                _ => unreachable!(),
+            }
+        }
+
+        None
+    }
+
+    /// GenericTypeArgs = ( '<' GenericTypeArgsList? '>' )
+    ///
+    /// GenericTypeArgsList = GenericTypeArg ( ',' GenericTypeArg )* ','?
+    ///
+    /// GenericTypeArg
+    ///     = NamedTypeArg
+    ///     | OrderedTypeArg
+    ///
+    /// NamedTypeArg = identifier '=' Type
+    ///
+    /// OrderedTypeArg = TypeOrTypeExpression
+    pub(super) fn parse_generic_type_args(&mut self) -> GenericTypeArgs {
+        let mut generic_type_args = GenericTypeArgs::default();
+        if !self.eat_less() {
+            return generic_type_args;
+        }
+
+        let generics = self.parse_many(
+            "generic parameters",
+            separated_by_comma().until(Token::Greater),
+            Self::parse_generic_type_arg,
+        );
+
+        for generic in generics {
+            match generic {
+                GenericTypeArg::Ordered(typ) => {
+                    generic_type_args.ordered_args.push(typ);
+                }
+                GenericTypeArg::Named(name, typ) => {
+                    generic_type_args.named_args.push((name, typ));
+                }
+            }
+        }
+
+        generic_type_args
+    }
+
+    fn parse_generic_type_arg(&mut self) -> Option<GenericTypeArg> {
+        if matches!(self.token.token(), Token::Ident(..)) && self.next_is(Token::Assign) {
+            let ident = self.eat_ident().unwrap();
+
+            self.eat_assign();
+
+            let typ = self.parse_type_or_error();
+            return Some(GenericTypeArg::Named(ident, typ));
+        }
+
+        // Otherwise
+        let typ = self.parse_type_or_type_expression();
+        let Some(typ) = typ else {
+            self.expected_label(ParsingRuleLabel::TypeOrTypeExpression);
+            return None;
+        };
```

Comment:
```suggestion
        let Some(typ) = self.parse_type_or_type_expression() else {
            self.expected_label(ParsingRuleLabel::TypeOrTypeExpression);
            return None;
        };
```

Reply:
I'm applying these one by one because for a batch GitHub tells me to go to the files tab, and it's kind of hard to find these comments in that big diff 😅